### PR TITLE
feat(crux): reworked email and notification templating

### PR DESCRIPTION
**Pull request by robot9706**
- Removed hardcoded notification and email texts from code and moved them to template files
- Implemented a simple template file
- Moved email and chat templates to a new `assets` folder
- Moved the install scripts from the root folder to the `assets` folder as well
- Fixed an issue with the node script generation form: selecting PowerShell for Docker, then selecting Kubernetes tries to generate a PS1 script for Kubernetes (which is not supported)

❓`assets` folder, template format (inplace + hbr file) /docs for this?/, `notification.template.builder.ts` service (naming?)

### CODE REVIEW

File: web/crux-ui/src/components/nodes/dyo-node-setup.tsx [hunk 1]
```diff
@@ -113,6 +113,10 @@ const DyoNodeSetup = (props: DyoNodeSetupProps) => {
             converter={(it: NodeType) => t(`technologies.${it}`)}
             onSelectionChange={it => {
               formik.setFieldValue('type', it, true)
+              if (it === 'k8s') {
```
Comment (m8vago) [inline web/crux-ui/src/components/nodes/dyo-node-setup.tsx:116]: we should move this if to the onTypeChanged callback, if that's possible

File: web/crux/src/builders/notification.template.builder.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+type TemplateFile = { [key: string]: TemplateEntry }
+
+export type TemplateType = 'email' | 'chat'
+
+export type ProcessedTemplate = { [key: string]: string }
+
+@Injectable()
+export default class NotificationTemplateBuilder {
+  private processTemplateEntry<TData>(entry: TemplateEntry, data: TData): string {
```
Comment (m8vago) [inline web/crux/src/builders/notification.template.builder.ts:21]: simply use T or TemplateType, please

**Review by polaroi8d — CHANGES_REQUESTED** [review]

File: web/crux/src/builders/notification.template.builder.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    if (!templateText) {
+      throw new InvalidArgumentException({
+        property: 'template',
+        value: 'undefined',
+        message: `Invalid template`,
+      })
+    }
+
+    return Handlebars.compile(templateText)(data)
```
Comment (polaroi8d) [inline web/crux/src/builders/notification.template.builder.ts:35]: "I am not very familiar with Handlebars, but what if it throws an exception? Isn't it necessary to have:

```
    try {
      return Handlebars.compile(templateText)(data);
    } catch (error) {
      throw new Error(`Could not compile template: ${error.message}`);
    }
    ```

File: web/crux/assets/chat/invite.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
```
[the remark this answers is not in the document]
Comment (robot9706) [reply web/crux/assets/chat/invite.json:1]: Works for me (`notificationTemaple` 😉 ).

**Review by m8vago — APPROVED** [review]

**Review by polaroi8d — APPROVED** [review]
LGTM

**Pull request merged**